### PR TITLE
[Refactor] update `metainfo` in pkl file and add `categories` into `metainfo`

### PR DESCRIPTION
**Pull request by ZCMax**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

unify the `metainto` of all datasets info file and add `categories` and `info_version` into `metainto`.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by Tai-Wang — APPROVED** [review]

File: tools/dataset_converters/update_infos_to_v2.py [hunk 1]
```diff
@@ -995,8 +1040,17 @@ def update_waymo_infos(pkl_path, out_dir):
...
-    converted_data_info = dict(
-        metainfo={'DATASET': 'Waymo'}, data_list=converted_list)
+
+    # dataset metainfo
+    metainfo = dict()
+    metainfo['categories'] = {k: i for i, k in enumerate(METAINFO['CLASSES'])}
+    if ignore_class_name:
+        for ignore_class in ignore_class_name:
+            metainfo['categories'][ignore_class] = -1
+    metainfo['dataset'] = 'waymo'
```
Comment (JingweiZhang12) [inline tools/dataset_converters/update_infos_to_v2.py:1050]: better to add version info for  waymo dataset.

**Review by JingweiZhang12 — APPROVED** [review]

**Review by VVsssssk — APPROVED** [review]

**Comment by VVsssssk** [comment]
If we reserve the class2labels map in the pkl file, I think we need to update kittimetric to dynamic convert classes label.

File: mmdet3d/evaluation/metrics/kitti_metric.py [hunk 2]
```diff
@@ -74,25 +74,19 @@ def __init__(self,
...
-            classes (list[str]): Classes used in the dataset. Default used
-                ['Pedestrian', 'Cyclist', 'Car', 'Van', 'Truck',
-                'Person_sitting', 'Tram', 'Misc'].
+            data_infos (dict): Data infos including metainfo and annotations
+                loaded from ann_file.
 
         Returns:
             List[dict]: List of Kitti annotations.
         """
+        cat2label = data_infos['metainfo']['categories']
```
Comment (ZwwWayne) [inline mmdet3d/evaluation/metrics/kitti_metric.py:87]: metainfo should not come from data_infos but from the dataset['metainfo']?

Comment (ZCMax) [reply mmdet3d/evaluation/metrics/kitti_metric.py:87]: the dataset['metainfo'] does not include the whole classes information. We need the whole classes info convert the pkl annotations back to Kitti-format annotations

**Review by ZwwWayne — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**